{
	"id": "CVE-2022-41902",
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2022-41902",
				"description": "TensorFlow is an open source platform for machine learning. The function MakeGrapplerFunctionItem takes arguments that determine the sizes of inputs and outputs. If the inputs given are greater than or equal to the sizes of the outputs, an out-of-bounds memory read or a crash is triggered. We have patched the issue in GitHub commit a65411a1d69edfb16b25907ffb8f73556ce36bb7. The fix will be included in TensorFlow 2.11.0. We will also cherrypick this commit on TensorFlow 2.8.4, 2.9.3, and 2.10.1.",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "nvd@nist.gov",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:H",
							"base_score": 9.1,
							"base_severity": "CRITICAL",
							"temporal_score": 9.1,
							"temporal_severity": "CRITICAL",
							"environmental_score": 9.1,
							"environmental_severity": "CRITICAL"
						}
					},
					{
						"type": "cvss_v31",
						"source": "security-advisories@github.com",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
							"base_score": 7.1,
							"base_severity": "HIGH",
							"temporal_score": 7.1,
							"temporal_severity": "HIGH",
							"environmental_score": 7.1,
							"environmental_severity": "HIGH"
						}
					}
				],
				"cwe": [
					{
						"source": "nvd@nist.gov",
						"cwe": [
							"CWE-125"
						]
					},
					{
						"source": "security-advisories@github.com",
						"cwe": [
							"CWE-787"
						]
					}
				],
				"references": [
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/utils/functions.cc#L221"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/commit/a65411a1d69edfb16b25907ffb8f73556ce36bb7"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-cg88-rpvp-cjv5"
					},
					{
						"source": "nvd.nist.gov",
						"url": "https://nvd.nist.gov/vuln/detail/CVE-2022-41902"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/utils/functions.cc#L221"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/commit/a65411a1d69edfb16b25907ffb8f73556ce36bb7"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-cg88-rpvp-cjv5"
					}
				],
				"published": "2022-12-06T22:15:10.513Z",
				"modified": "2024-11-21T07:24:01.86Z"
			},
			"segments": [
				{
					"ecosystem": "cpe"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "cpe",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterias": [
							{
								"operator": "OR",
								"criterias": [
									{
										"operator": "OR",
										"criterias": [
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.8.4"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.9.3",
																		"ge": "2.9.0"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"cpe": "cpe:2.3:a:google:tensorflow:2.10.0:-:*:*:*:*:*:*"
															}
														}
													}
												]
											}
										]
									}
								]
							}
						]
					}
				}
			]
		}
	],
	"data_source": {
		"id": "nvd-api-cve",
		"raws": [
			"vuls-data-raw-nvd-api-cve/2022/CVE-2022-41902.json"
		]
	}
}
